### PR TITLE
Added measuring solving time.

### PR DESCRIPTION
**Pull request by kobrineli**
Added measuring solving time to switch between inputs in FuzzBench depending on summary solving time.

### CODE REVIEW

File: src/libtriton/api/api.cpp [hunk 1]
```diff
@@ -1089,15 +1089,15 @@ namespace triton {
   }
```
Comment (SweetVishnya) [inline src/libtriton/api/api.cpp:1089]: isSat function + docs

File: src/libtriton/engines/solver/z3/z3Solver.cpp [hunk 2]
```diff
@@ -136,6 +140,12 @@ namespace triton {
             /* Decrement the limit */
             limit--;
           }
+
+          /* Get time of solving end */
+          auto end = std::chrono::system_clock::now();
+
+          if (solving_time)
+            *solving_time = (std::chrono::duration_cast<std::chrono::milliseconds>(end - start)).count();
```
Comment (SweetVishnya) [inline src/libtriton/engines/solver/z3/z3Solver.cpp:148]: redundant parentheses

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: src/libtriton/api/api.cpp (hunk 1 above)
Comment (kobrineli) [reply src/libtriton/api/api.cpp:1089]: Done

File: src/libtriton/engines/solver/z3/z3Solver.cpp (hunk 2 above)
Comment (kobrineli) [reply src/libtriton/engines/solver/z3/z3Solver.cpp]: Done

**Comment by JonathanSalwan** [comment]
Can you also update the Python Bindings and add an unit test?

File: src/libtriton/api/api.cpp [hunk 3]
```diff
@@ -1089,15 +1089,15 @@ namespace triton {
...
 
 
-  std::unordered_map<triton::usize, triton::engines::solver::SolverModel> API::getModel(const triton::ast::SharedAbstractNode& node, triton::engines::solver::status_e* status, triton::uint32 timeout) const {
+  std::unordered_map<triton::usize, triton::engines::solver::SolverModel> API::getModel(const triton::ast::SharedAbstractNode& node, triton::engines::solver::status_e* status, triton::uint32 timeout, triton::uint32 *solving_time) const {
```
Comment (JonathanSalwan) [inline src/libtriton/api/api.cpp:1092]: Can you replace all `triton::uint32 *var_name` by `triton::uint32* var_name`? The pointer keyword (`*`) must be attached to its type in matter of following the coding style :)

**Comment by SweetVishnya** [comment]
AFAIK, @kobrineli was working on Python bindings and tests. Anyway, @JonathanSalwan you may (or not) add them as you wish. Moreover, we should mind the same parameter for Bitwuzla in #1062

**Comment by JonathanSalwan** [comment]
Mmmh, as i got no answer I'm currently deving the Python bindings but if he's working on it i will let him finish :)

**Comment by SweetVishnya** [comment]
If you are already working, then you would make it faster. @kobrineli is on his exams and won't be able to commit soon.

P.S. We should speak up plans on our side in future)))

**Comment by JonathanSalwan** [comment]
> If you are already working, then you would make it faster. @kobrineli is on his exams and won't be able to commit soon.

Ok I can do that :). Good luck to him for his exams.

> P.S. We should speak up plans on our side in future)))

Yeah, because I see an army of the ISP RAS guys committing on Triton :)

**Pull request merged**